### PR TITLE
Initial spec doc contribution

### PR DESCRIPTION
**Pull request by tivrfoa**
Hi,

For sure there are things to improve and fix, but I think it's good enough for an initial contribution.

### CODE REVIEW

File: spec/src/main/asciidoc/messaging-spec.adoc [hunk 1]
```diff
@@ -26,3 +26,7675 @@ include::license-efsl.adoc[]
...
+|TopicPublisher |NO
+|TopicSubscriber |NO
+|===
+
+Jakarta Messaging defines some specific rules that restrict the 
+concurrent use of sessions. These apply to the Session object in the 
+classic API and to the QueueSession and TopicSession objects in the 
+domain-specific APIs. They also apply to the JMSContext object in the 
+simplified API since it encompasses a session. Since these rules require
+ more knowledge of Jakarta Messagingspecifics than we have presented at
```
Comment (bshannon) [inline spec/src/main/asciidoc/messaging-spec.adoc:629]: There's a gratuitous spec at the beginning of the line and a missing space in the middle between two words.

File: spec/src/main/asciidoc/messaging-spec.adoc [hunk 2]
```diff
@@ -26,3 +26,7675 @@ include::license-efsl.adoc[]
...
+=== Request/reply
+
+Jakarta Messaging provides the JMSReplyTo message header field for 
+specifying the Destination where a reply to a message should be sent. 
+The JMSCorrelationID header field of the reply can be used to reference 
+the original request. See Section “” for more information.
+
+In addition, Jakarta Messaging provides a facility for creating 
+temporary queues and topics that can be used as a unique destination for
+ replies.
```
Comment (bshannon) [inline spec/src/main/asciidoc/messaging-spec.adoc:667]: Another gratuitous space at the beginning.  Is this the result of the line length tool?

File: spec/src/main/asciidoc/messaging-spec.adoc [hunk 3]
```diff
@@ -26,3 +26,7675 @@ include::license-efsl.adoc[]
...
+Within this general form, the definition of a message varies
+significantly across products. There are major differences in the
+content and semantics of headers. Some products use a self describing,
+canonical encoding of message data; others treat data as completely
+opaque. Some products provide a repository for storing message
+descriptions that can be used to identify and interpret message content;
+others don’t.
+
+It would be quite difficult for Jakarta Messaging to capture the breadth
+ of this,sometimes conflicting, union of message models.
```
Comment (bshannon) [inline spec/src/main/asciidoc/messaging-spec.adoc:698]: And another.  You'll need to search the entire document to find them all.

File: spec/src/main/asciidoc/messaging-spec.adoc [hunk 4]
```diff
@@ -26,3 +26,7675 @@ include::license-efsl.adoc[]
...
+
+==== JMSDeliveryMode
+
+The JMSDeliveryMode header field contains the delivery mode specified
+when the message was sent.
+
+When a message is sent this value is ignored. After completion of the
+send, it holds the delivery mode specified by the sending method.
+
+See Section  “” for more information.
```
Comment (bshannon) [inline spec/src/main/asciidoc/messaging-spec.adoc:759]: Missing reference?

File: spec/src/main/asciidoc/messaging-spec.adoc (hunk 2 above)
Comment (tivrfoa) [reply spec/src/main/asciidoc/messaging-spec.adoc:667]: > Another gratuitous space at the beginning. Is this the result of the line length tool?

Probably. Thanks a lot for reviewing it.

**Comment by tivrfoa** [comment]
> The changes look good except for the main document. The main document is impossible
> to review as is because the diffs show that everything is new. Did you not check in the
> original text from the CQ before starting your changes? That makes it hard to tell what
> you fixed.
> 
> Also, if the line length tool is going to touch lots of lines, that will make it hard as well.
> Line length doesn't look to be a problem in the original contribution so I'm not sure
> why you think you need that tool.
> 
> Also, it look like that tool is introducing some errors. There's a bunch of places where
> there's a gratuitous space at the beginning of the line and a missing space within the
> line. Fixing all of those by hand will be a pain.
> 
> My advice is to first check in the original text, then apply your changes, and skip the
> line length tool.

I reset to the commit previous to the changes to the line size.

The content of messaging-spec.adoc in master branch only has a few lines.
I then appended the "original" content o the zip file in the commit below:
https://github.com/eclipse-ee4j/jms-api/pull/267/commits/7a4143d99dd45b0da7ecfda79aefbcdd0f9c8628

File: spec/src/main/asciidoc/messaging-spec.adoc (hunk 4 above)
Comment (tivrfoa) [reply spec/src/main/asciidoc/messaging-spec.adoc:750]: This reference was already missing in the original document.

**Comment by tivrfoa** [comment]
> Line length doesn't look to be a problem in the original contribution so I'm not sure
> why you think you need that tool.
> 
I thought it could be a problem because replacing JMS for Jakarta Messaging introduces 14 new characters.

File: spec/src/main/asciidoc/messaging-spec.adoc (hunk 4 above)
Comment (tivrfoa) [reply spec/src/main/asciidoc/messaging-spec.adoc:750]: > Missing reference?

The original Messaging.adoc also does not have this reference ... But the file Messaging21.docx seems to have them =D

Would you like me to include them in this pull request?

**Comment by OndroMih** [comment]
Hi @tivrfoa , @bshannon  is right, it would be easier to review if your first committ contained only the original version donated in the CQ and all the other commits with changes are rebased on top of that. I think it can be within the same PR as it's possible to see the changes done in individual commits.

**Comment by tivrfoa** [comment]
> Hi @tivrfoa , @bshannon is right, it would be easier to review if your first committ contained only the original version donated in the CQ and all the other commits with changes are rebased on top of that. I think it can be within the same PR as it's possible to see the changes done in individual commits.

Hi @OndroMih . It was already done like this. Please see this reply:
https://github.com/eclipse-ee4j/jms-api/pull/267#issuecomment-606962844

The commit below introduces the original version:
 Append original specification (Messaging.adoc) to messaging-spec.adoc 7a4143d 

The other commits add changes to it.

**Comment by bshannon** [comment]
How do I see the diff between the original and the updates and then make comments
on the updates?

**Comment by tivrfoa** [comment]
> How do I see the diff between the original and the updates and then make comments
> on the updates?

You are able to comment on each commit.
The first commit that changes the original file is this one:
https://github.com/eclipse-ee4j/jms-api/pull/267/commits/7a4143d99dd45b0da7ecfda79aefbcdd0f9c8628

**Comment by OndroMih** [comment]
Thanks, @tivrfoa.

To make it easier to see the differences, I've created a pull request in my forked repo to show the differences between the commit of the original doc and the last commit in this pull request #267. Here you can see the diff: https://github.com/OndroMih/jms-api/pull/1/files, @bshannon

**Comment by bshannon** [comment]
Why did you end up closing this?  Should I no longer review it?

**Comment by tivrfoa** [comment]
> Why did you end up closing this? Should I no longer review it?

Ondo created a new pull request. Thank you! :)

**Comment by bshannon** [comment]
Where's the new PR?

**Comment by tivrfoa** [comment]
> Where's the new PR?

https://github.com/OndroMih/jms-api/pull/1/files

**Comment by tivrfoa** [comment]
> Where's the new PR?

I think maybe I messed up :/ thought he had created another one ...
But that is ok because Ondo has a branch with the changes I made...
I thought I could help but I don't know enough. Good luck guys.

**Comment by bshannon** [comment]
This new PR isn't against the jms-api repo, so when we approve and merge it, it won't change
that repo at all.  Are we supposed to assume that approving this PR will have the intended
effect on the jms-api repo, through some magic?  Are you just going to apply all the changes
to this fork of jms-api, then apply a PR with all these changes from this forked repo to the
jms-api repo?

I think this will work, but you need to explain to your reviewers what it is you doing and what
you expect them to do.

**Comment by OndroMih** [comment]
Hi @tivrfoa, I'm sorry for confusion but my pull request shouldn't replace your pull request. My PR is only for the sake of displaying the differences between the original document (in your first commit) and the document after your changes (your last commit) so that it's easier to review your pull request. Please reopen this pull request so that we can continue reviewing it.

**Comment by OndroMih** [comment]
I've closed my pull request because it's not meant to be merged. It can still be used to see the changes against the original donated document. But we should really reopen this pull request try to reach a point when we can merge it.

**Comment by bshannon** [comment]
This can't be reopened because the repository from which it was submitted has been deleted.
Someone needs to start over with a new pull request.  @OndroMih your pull request could be
used to create a new pull request, although you'll need to do it "manually".

**Comment by OndroMih** [comment]
OK, I'll see what I can do.

**Pull request closed without merging**